types/spec/legend.d.ts

```diff
@@ -28,105 +31,294 @@ export type LegendOrient =
   | 'bottom-left'
   | 'bottom-right';
 
-export type Color = string;
+export type LegendDirection = 'vertical' | 'horizontal';
 
-export interface BaseLegend {
-  name?: string;
+export interface Legend extends BaseLegendConfig {
```

Comment:
Hmm but a lot of BaseLegendConfig should support signal too, so need to use some Record or something like `{[k in BaseLegendConfig]: BaseLegendConfig[K] | SignalRef}`?